fact_extractor/plugins/unpacking/generic_carver/code/generic_carver.py

```diff
@@ -22,8 +24,50 @@ def unpack_function(file_path, tmp_dir):
     output = execute_shell_command('binwalk --extract --carve --signature --directory  {} {}'.format(tmp_dir, file_path))
 
     drop_underscore_directory(tmp_dir)
+    screening_meta = remove_false_positive_archives(file_path, tmp_dir)
 
-    return {'output': output}
+    return {'output': output, 'screening': screening_meta}
+
+
+def remove_false_positive_archives(original_filename: str, unpack_directory: str) -> str:
+    binwalk_root = Path(unpack_directory) / f'_{original_filename}.extracted'
+    if not binwalk_root.exists() or not binwalk_root.is_dir():
+        return 'No files extracted, so nothing removed'
+    screening_log = ''
+
+    for file_path in binwalk_root.iterdir():
+        file_type = magic.from_file(str(file_path), mime=True)
+
+        if 'zip' in file_type:
+            output_unzip = execute_shell_command('unzip -l {}'.format(file_path))
+            if 'not a zipfile' in output_unzip.replace('\n ', ''):
+                os.remove(file_path)
+                screening_log = 'file was not a zipfile and was removed'
```

Comment:
- the output gets overwritten if there are multiple files that get removed
    - you could add the logging strings to a list and do a `'\n'.join()` at the end
- it could be informative to include the file name into the log